### PR TITLE
chore(ci): add ci and release workflows

### PR DESCRIPTION
**Pull request by hongaar**
This removes the Travis integration and adds two GitHub actions workflows:

- A CI workflow replacing Travis (build & test)
- A release workflow using `semantic-release` which publishes to NPM on each commit to master

## Todo

- [x] Migrate from codecov to codeclimate using [codeclimate-action](https://github.com/paambaati/codeclimate-action)
- [ ] Add https://github.com/apps/semantic-pull-requests

### CODE REVIEW

**Comment by hongaar** [comment]
See example run here: https://github.com/hongaar/vorpal/commit/f92e430bf537582939534a8e47a275d056f3b56c/checks?check_suite_id=319222394

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    needs: test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - run: npm ci
+      - run: npm run compile
+      - uses: paambaati/codeclimate-action@v2.3.0
```
Comment (AdrieanKhisbe) [inline .github/workflows/ci.yml:29]: why dropping `codecov` for `codeclimate`?

**Review by AdrieanKhisbe — CHANGES_REQUESTED** [review]
I'm not against trying out github actions, for test, and maybe even publication

However I think it's a bit premature to drop travis.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (hongaar) [reply .github/workflows/ci.yml:29]: Codecov PR comments are a bit intrusive imho (but then maybe they can be disabled?)

Code Climate has additional code quality checks (duplication, complexity, etc) on top of two different coverage checks (a coverage threshold (%) for changed code and an overall coverage diff).

Comment (hongaar) [reply .github/workflows/ci.yml:29]: I'm okay with reverting back to Codecov, if we can get rid of the comments 😉

**Comment by Ore4444** [comment]
I suggest, to simplify discussion, if we're pro using GitHub Actions, make 1 PR to recreates the same Travis config as GH actions, and let's merge that. Then we can separately discuss modifications to the CI

**Comment by hongaar** [comment]
Makes sense, will split this PR in two (will need the codecov token, @AdrieanKhisbe did you set it up?)

_Edit, split in three: #24, #25, #26_

**Comment by AdrieanKhisbe** [comment]
@hongaar about the token there is any.
Travis Gilly integrate with codecov (or thé reverse), si there is no nées while using travis

**Pull request closed without merging**